### DISCUSSION TITLE
Is there a way to use custom providers for DoT

### DISCUSSION THREAD

**Discussion by spar01** [qdm12/gluetun#679] [Q&A]
Per the Wiki there is an env var to select a few DoT providers. I am wondering if there is a way to use a custom DoT provider?
Specifically looking at using Mullvad's DoT service.
For reference: https://mullvad.net/en/help/dns-over-https-and-dns-over-tls/

**Comment by qdm12** [comment]
Please create an issue for that on https://github.com/qdm12/dns and it'll be added (in the coming weeks) to gluetun.

I'm still working on the v2 of that DNS program & library, which will be used in gluetun once finalized. For now it all works, except DNSSEC validation, which is the last critical piece before it can replace the existing dns implementation (unbound).

So as a consequence, all DNS related issues are stacking up on that v2 (so I don't go too crazy maintaining 2 versions 😄)

**Reply by spar01** [reply]
Thanks, I'll open an issue on there.

I don't have much networking knowledge so bear with me:
As for the current state of gluetun if DoT is disabled and DNS_PLAINTEXT_ADDRESS is set, does the DNS query go out unencrypted from the local network or from the VPNs network?

**Reply by Magic-Deceiver** [reply]
Following because I am interested as well!
